src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/AuditUtility.cs

```diff
@@ -152,11 +153,19 @@ private static (string severityLabel, NuGetLogCode code) GetSeverityLabelAndCode
             Dictionary<PackageIdentity, Dictionary<PackageVulnerabilityInfo, List<string>>>? result = null;
 
             int minSeverity = ParseAuditLevel();
+            Func<RestoreTargetGraph, IEnumerable<ResolvedDependencyKey>> getPackagesToCheck =
+                ParseAuditMode() switch
+                {
+                    NuGetAuditMode.Direct => GetDirectDependencies,
+                    NuGetAuditMode.Transitive => GetAllDependencies,
+                    _ => GetDirectDependencies
+                };
 
-            foreach (var graph in _targetGraphs)
+            foreach (RestoreTargetGraph graph in _targetGraphs)
             {
-                foreach (LibraryIdentity package in graph.Flattened.Select(i => i.Key).Where(p => p.Type == LibraryType.Package))
+                foreach (ResolvedDependencyKey resolvedDependency in getPackagesToCheck(graph))
```

Comment:
Could you please confirm if we're parsing audit mode for every target framework graph? Assuming that's the case, and since audit mode is set at the project level, I believe there's no need to fetch it in the nested foreach loop.

Reply:
Audit mode is parsed in the method `ParseAuditMode`, which you can see if called outside any `foreach` loop. So, it's only parsed once per project.

Reply:
Thanks Andy, the syntax for hooking the delegate handler was new to me. Hence wanted to confirm the behavior.